### DISCUSSION TITLE
Looking for implementation of GCConfig::GetBGCSpinCount

### DISCUSSION THREAD

**Discussion by RobertGBryan** [dotnet/runtime#58883] [Q&A] [answered]
I was exploring [GC.cpp](https://raw.githubusercontent.com/dotnet/coreclr/master/src/gc/gc.cpp) and trying to figure out how often the sleep method is called inside of the method allocate_small. A variable called bgc_alloc_spin_count is first initialized to 140, but then is reinitialized to the return value of GCConfig::GetBGCSpinCount when BACKGROUND_GC is defined, which it usually will be since it is the default. This occurs during when the GC is first initialized.

I then found the GCConfig.h and GCConfig.cpp [files here](https://github.com/dotnet/runtime/tree/main/src/coreclr/gc). But neither file has a definition or implementation of GetBGCSpinCount. 

So, is this a versioning issue? Maybe GC.cpp is old and out of date? If not, where can I find the definition of GetBGCSpinCount?

**Comment by CyrusNajmabadi** [comment]
Moving to runtime

**Comment by RobertGBryan** [comment, marked as answer]
The code is using a C++ macro which ends up generating the method name. This can be seen in GCConfig.h and GCConffig.cpp.
